### PR TITLE
(mu, lambda)-CMA-ES optimizer implementation 

### PR DESCRIPTION
**Pull request by jpzk**
Here's a (mu, lambda)-CMA-ES optimizer implementation. I had to modifiy the arg creator class in order to convert between numpy arrays and the MetaOpt specific domain model (with args). It is working for me. 
The mutation strength is global and is fixed to 0.5. We have to discuss how to handle meta parameters like mutation strength (in the meeting). Please review and add comments.

### CODE REVIEW

File: metaopt/optimizer/cmaes.py [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+
+	# initialize the parameters with member variables
+	self.initialize_parameters()
+
+        while not self.exit_condition():
+            self.add_offspring()
+            self.score_population()
+
+            if self.aborted:
+                return self.best_scored_indivual[0][0]
```
Comment (renke) [inline metaopt/optimizer/cmaes.py]: I think this should be  `return self.best_scored_indivual[0]`.

File: metaopt/optimizer/cmaes.py [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+
+            self.select_parents()
+
+            self.generation += 1
+
+        return self.best_scored_indivual[0][0]
```
Comment (renke) [inline metaopt/optimizer/cmaes.py]: See above.

**Comment by jpzk** [comment]
fixed it. :)

**Comment by renke** [comment]
We will merge this when a bug with wait() is fixed. Currently, it only calls the objective function for the individuals in the initial population.

**Comment by renke** [comment]
Can you change the optimizer so that is respects the the number type of the parameter and its constraints?

You could raise an exception when the objective function uses a boolean or integer parameter type (the types are specified in in the `param_spec` object passed to the `optimize` method). At least you should respect the given interval for a float parameter (ignoring the step size is okay for now, I guess). Currently the optimizer will happily use values that are not in the specified parameter interval.

**Comment by jpzk** [comment]
The number type is not respected in the other optimizers as well. Therefore, I cant find this specific exception. Regarding interval constraints, other optimizers dont care as well. Generally, optimizing with respect to a certain interval is a bit complicated, because it introduces linear constraints, which require constraint handling such as death penalty or the repairing of solutions. We need to discuss this in more detail.

**Comment by renke** [comment]
The number type of parameters is actually implicitly respected by all existing optimizers, because they use the classes ArgsModifier and ArgsCreator (and thus the methods from the Arg class and it's subclasses). They ensure an argument retains its type and is within its defined interval. Let's discuss this topic on the next meeting though.

**Pull request merged**